device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/fileupload/FileUploadTask.java

```diff
@@ -181,12 +182,29 @@ public FileUploadSasUriResponse getFileUploadSasUri(FileUploadSasUriRequest requ
         responseMessage = httpsTransportManager.getFileUploadSasUri(message);
         httpsTransportManager.close();
 
-        if (responseMessage.getBytes() == null || responseMessage.getBytes().length == 0)
+        String responseMessagePayload = null;
+        if (responseMessage.getBytes() != null && responseMessage.getBytes().length > 0)
+        {
+            responseMessagePayload = new String(responseMessage.getBytes(), DEFAULT_IOTHUB_MESSAGE_CHARSET);
+        }
+
+        IotHubServiceException serviceException =
+            IotHubStatusCode.getConnectionStatusException(
+                responseMessage.getStatus(),
+                responseMessagePayload);
+
+        // serviceException is only not null if the provided status code was a non-successful status code like 400, 429, 500, etc.
+        if (serviceException != null)
+        {
+            throw new IOException("Failed to get the file upload SAS URI", serviceException);
```

Comment:
This method only declares that is throws IOExceptions, so the closest thing we can do to throwing an IotHubServiceException is to nest it within an IOException.